examples/twittermap/web/public/javascripts/lib/webgl_point_layer.js

```diff
@@ -0,0 +1,361 @@
+        this._programs[0].selectedLoc = gl.getUniformLocation(this._programs[0], "u_selected");
+        this._programs[0].vertLoc = gl.getAttribLocation(this._programs[0], "a_vertex");
+        this._programs[0].indexLoc = gl.getAttribLocation(this._programs[0], "a_index");
+
+        var fragmentShader2 = gl.createShader(gl.FRAGMENT_SHADER);
+        gl.shaderSource(fragmentShader2, fragCode2);
+        gl.compileShader(fragmentShader2);
+        if (!gl.getShaderParameter(fragmentShader2, gl.COMPILE_STATUS)) {
+            alert(gl.getShaderInfoLog(fragmentShader2));
+        }
+
+        this._programs[1] = gl.createProgram();
+        gl.attachShader(this._programs[1], vertexShader);
+        gl.attachShader(this._programs[1], fragmentShader2);
+        gl.linkProgram(this._programs[1]);
+        if (!gl.getProgramParameter(this._programs[1], gl.LINK_STATUS)) {
+            alert("Could not initialise shaders");
+        }
+
+        gl.useProgram(this._programs[1]);
+
+        this._programs[1].matLoc = gl.getUniformLocation(this._programs[1], "u_matrix");
+        this._programs[1].pointSize = gl.getUniformLocation(this._programs[1], "u_pointSize");
+        this._programs[1].vertLoc = gl.getAttribLocation(this._programs[1], "a_vertex");
+        this._programs[1].indexLoc = gl.getAttribLocation(this._programs[1], "a_index");
+
+        gl.useProgram(null);
+
+        this._vertBuffer = gl.createBuffer();
+    },
+
+
+    _checkData: function(data) {
+        if ( data == null ) return false;
+        try {
+            for ( var i = 0; i < data.length; ++i )
+                if ( data[i].length !== 3 ) return false;
+            return true;
+
+        } catch (err) {
+            console.log(err);
+            return false;
+        }
+    },
+
+
+    _initBuffers: function() {
+        var gl = this._gl;
+        if ( gl == null || this._data == null ) return;
+
+        var verts = [];
+        for ( var id = 1; id <= this._data.length; ++id ) {
+            var pixel = this._LatLongToPixel_XY(this._data[id - 1][0], this._data[id - 1][1]);
+
+            ///// id = r + 256*g + 256^2*b
+            var r = Math.floor(id / (65536));
+            var g = Math.floor((id % (65536)) / 256);
+            var b = id % 256;
+
+            verts.push(pixel.x, pixel.y, r, g, b);
+        }
+
+        this._vertArray = new Float32Array(verts);
+        var fsize = this._vertArray.BYTES_PER_ELEMENT;
+
+        gl.bindBuffer(gl.ARRAY_BUFFER, this._vertBuffer);
+        gl.bufferData(gl.ARRAY_BUFFER, this._vertArray, gl.STATIC_DRAW);
+
+        gl.useProgram(this._programs[0]);
+        gl.vertexAttribPointer(this._programs[0].vertLoc, 2, gl.FLOAT, false, fsize*5, 0);
+        gl.enableVertexAttribArray(this._programs[0].vertLoc);
+        gl.vertexAttribPointer(this._programs[0].indexLoc, 3, gl.FLOAT, false, fsize*5, fsize*2);
+        gl.enableVertexAttribArray(this._programs[0].indexLoc);
+
+        gl.useProgram(this._programs[1]);
```

Comment:
1. Perhaps we can use another name rather than programs[0] and [1] which are not clear to me;
2. We can add a separate method to load the programs[0] and [1] to avoid copying the codes.